cmd/auth-handler.go

```diff
@@ -125,30 +145,62 @@ func checkAdminRequestAuthType(r *http.Request, region string) APIErrorCode {
 	return s3Err
 }
 
-func checkRequestAuthType(ctx context.Context, r *http.Request, action policy.Action, bucketName, objectName string) APIErrorCode {
-	isOwner := true
-	accountName := globalServerConfig.GetCredential().AccessKey
+// Fetch the security token set by the client.
+func getSessionToken(r *http.Request) (token string) {
+	token = r.Header.Get("X-Amz-Security-Token")
+	if token != "" {
+		return token
+	}
+	return r.URL.Query().Get("X-Amz-Security-Token")
+}
+
+// Fetch claims in the security token returned by the client and validate the token.
+func getClaimsFromToken(r *http.Request, claims map[string]interface{}) APIErrorCode {
```

Comment:
Ah, finally understood what it is doing after poring over Parse, ParseWithClaims, and ParseUnverified in jwt-go. It is more appropriate to let the function itself allocate the claims:

```
func getClaimsFromToken(r *http.Request) (map[string]interface{}, APIErrorCode) {}
```

Moreover, you can use `p.Parse` instead of `p.ParseWithClaims` as that already parses (and allocates) claims and stores it in `jtoken.Claims`.